include/NotificationsManager.php

```diff
@@ -827,4 +828,38 @@ private function formatIntros($intros) {
 
 		return $arr;
 	}
+
+	/**
+	 * @brief Check for missing contact data and try to fetch the data from 
+	 *     from other sources
+	 * 
+	 * @param array $arr The input array with the intro data
+	 * 
+	 * @return array The array with the intro data
+	 */
+	private function getMissingIntroData($arr) {
+		// If the network and the addr isn't available from the gcontact
+		// table entry, take the one of the contact table entry
+		if ($arr['gnetwork'] == "") {
+			$arr['gnetwork'] = $arr['network'];
+		}
+		if ($arr['gaddr'] == "") {
+			$arr['gaddr'] = $arr['addr'];
+		}
+
+		// If the network and addr is still not available try to probe
+		// the contact url to fetch the missing data
+		if ($arr['gnetwork'] == "" || $arr['gaddr'] == "") {
+			$ret = Probe::uri($arr["url"]);
```

Comment:
I guess you could call "get_contact_details_by_url" instead of the probing here - since the function is doing a probing itself as well. Later (in the 3.5.3) we could refine the whole process.